renderdoc/driver/ihv/amd/amd_counters.cpp

```diff
@@ -22,70 +22,202 @@
  * THE SOFTWARE.
  ******************************************************************************/
 
+#include <sstream>
+#include <iostream>
+#include <chrono>
+#include <thread>
 #include "amd_counters.h"
-#include <Windows.h>
 #include "common/common.h"
 #include "core/plugins.h"
 #include "official/GPUPerfAPI/Include/GPUPerfAPI.h"
 #include "official/GPUPerfAPI/Include/GPUPerfAPIFunctionTypes.h"
-
 #include "strings/string_utils.h"
 
+#if ENABLED(RDOC_WIN32)
+#include <Windows.h>
+#else
+#include <dlfcn.h>
+#endif
+
+inline bool AMD_FAILED(GPA_Status status)
+{
+  return status != GPA_STATUS_OK;
+}
+
+inline bool AMD_SUCCEEDED(GPA_Status status)
+{
+  return status == GPA_STATUS_OK;
+}
+
+static void GPA_LoggingCallback(GPA_Logging_Type messageType, const char* pMessage)
+{
+  std::string message;
+
+  switch (messageType)
```

Comment:
It ends up being redundant adding your own prefix here since it's then also prefixed with the message type by the pre-existing logging. If you wanted you could convert `GPA_LOGGING_ERROR` -> `RDCERR`, `GPA_LOGGING_MESSAGE` -> `RDCLOG` and `GPA_LOGGING_TRACE` -> `RDCDEBUG` but otherwise you should be able to print the message straight.